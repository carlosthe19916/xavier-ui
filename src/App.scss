// Importing Global Variables
@import "~@redhat-cloud-services/frontend-components-utilities/files/Utilities/_all";
@import "~@redhat-cloud-services/frontend-components/index.css";

.pf-c-page__main {
    background-color: #e7e7e7;
}
