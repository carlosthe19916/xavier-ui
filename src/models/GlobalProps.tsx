export interface GlobalProps {
    history: any;
    location: any;
    match: any;
}
