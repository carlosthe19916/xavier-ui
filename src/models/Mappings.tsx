export interface FlagAssessmentModel {
    flag: string;
    label: string;
    flagAssessmentOSModels: FlagAssessmentOSModel[]
}

export interface FlagAssessmentOSModel {
    osName: string | undefined;
    assessment: string;
}
