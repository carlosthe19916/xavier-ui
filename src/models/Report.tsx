export interface Report {
    id: number;
    customerId: string;
    fileName: string;
    numberOfHosts: number;
    totalDiskSpace: number;
    totalPrice: number;
    creationDate: number
}
