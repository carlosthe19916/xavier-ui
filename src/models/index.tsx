export * from './Report';
export * from './Upload';
export * from './User';
export * from './Mappings';
