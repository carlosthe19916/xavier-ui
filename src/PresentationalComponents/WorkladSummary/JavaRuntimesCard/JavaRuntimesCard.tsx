import React from 'react';
import { Stack, StackItem, Divider, Bullseye } from '@patternfly/react-core';
import { BullseyeIcon } from '@patternfly/react-icons';
import ReportCard from '../../ReportCard';
import { ReportWorkloadSummary, JavaRuntimeModel } from 'src/models';
import FancyChartDonut from '../../FancyChartDonut';
import { FancyChartDonutData } from '../../FancyChartDonut/FancyChartDonut';
import { formatNumber, formatPercentage } from '../../../Utilities/formatValue';
import { EmptyCard } from '../../EmptyCard';

interface Props {
    reportWorkloadSummary: ReportWorkloadSummary | null;
}

export const JavaRuntimesCard: React.FC<Props> = ({ reportWorkloadSummary }) => {
    const title = 'Oracle Java runtimes information';

    if (!reportWorkloadSummary) {
        return <EmptyCard title={title} />;
    }

    const javaRuntimes = reportWorkloadSummary.javaRuntimes;
    if (!javaRuntimes || javaRuntimes.length === 0) {
        return <EmptyCard title={title} />;
    }

    const orderedJavaRuntimes = javaRuntimes.sort((a: JavaRuntimeModel, b: JavaRuntimeModel) => {
        if (a.vendor === b.vendor) {
            return Number(a.version) - Number(b.version);
        }
        return a.vendor.localeCompare(b.vendor);
    });

    //
    const pieValues = orderedJavaRuntimes.map(element => element.total);

    const total = pieValues.reduce((a: number, b: number) => a + b, 0);
    const percentages = pieValues.map((val: number) => val / total);

    const chartProps = {
        title: formatNumber(total, 0),
        subTitle: 'Total',
        height: 270,
        width: 300
    };
    const chartLegendProps = {
        height: 200,
        width: 210,
        responsive: false,
        y: 60
    };

    const chartData: FancyChartDonutData[] = orderedJavaRuntimes.map((element, index: number) => ({
        label: element.vendor + ' ' + element.version,
        value: percentages[index],
        extraData: pieValues[index]
    }));

    const tickFormat = (label: string, value: number, data: any) => {
        return `${label}: ${data}`;
    };
    const tooltipFormat = ({ datum }) =>
        `${datum.x}: ${formatPercentage(datum.y, 2)} \n Runtimes: ${formatNumber(datum.extraData, 0)}`;

    return (
        <ReportCard title={title} skipBullseye={true}>
            <Stack>
                <StackItem>
                    <div className="pf-c-content">
                        <h5>
                            <i>
                                <BullseyeIcon />
                            </i>{' '}
                            {total} Oracle JDKs can be replaced with OpenJDK
                        </h5>
                        <br />
                        <Divider component="div" />
                    </div>
                </StackItem>
                <StackItem isFilled>
                    <Bullseye>
                        <FancyChartDonut
                            data={chartData}
                            chartProps={chartProps}
                            chartLegendProps={chartLegendProps}
                            tickFormat={tickFormat}
                            tooltipFormat={tooltipFormat}
                        />
                    </Bullseye>
                </StackItem>
            </Stack>
        </ReportCard>
    );
};
