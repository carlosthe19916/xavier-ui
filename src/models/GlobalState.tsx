import { Report } from './Report';
import { Upload } from './Upload';

export interface GlobalState {
    reports: {
        report: Report,
        reports: Report[];
        loading: boolean;
        error: string;
        total: number;
    },
    uploads: {
        uploads: Upload[]
    }
}
