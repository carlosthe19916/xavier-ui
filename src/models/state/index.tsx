import { Report } from '../Report';
import { Upload } from '../Upload';

export interface GlobalState {
    reports: ReportState,
    uploads: UploadState
}

export interface ReportState {
    error: string | null;
    total: number;
    report: Report | null;
    reports: Report[];
    loading: boolean;
}

export interface UploadState {
    uploads: Upload[]
}