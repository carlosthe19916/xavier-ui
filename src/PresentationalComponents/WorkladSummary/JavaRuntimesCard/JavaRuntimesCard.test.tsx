import React from 'react';
import { shallow } from 'enzyme';
import { JavaRuntimesCard } from './JavaRuntimesCard';
import { ReportWorkloadSummary } from 'src/models';

const BasicReportWorkloadSummary: ReportWorkloadSummary = {
    complexityModel: {
        easy: 1,
        hard: 2,
        medium: 3,
        unknown: 4,
        unsupported: 5
    },
    recommendedTargetsIMSModel: {
        ocp: 10,
        osp: 20,
        rhel: 30,
        rhv: 40,
        total: 100
    },
    scanRunModels: [],
    summaryModels: [],
    workloadsDetectedOSTypeModels: [],
    javaRuntimes: []
};

describe('JavaRuntimesCard', () => {
    it('expect to render', () => {
        const reportWorkloadSummary: ReportWorkloadSummary = {
            ...BasicReportWorkloadSummary,
            javaRuntimes: [
                {
                    vendor: 'VendorA',
                    version: '8',
                    total: 10
                },
                {
                    vendor: 'VendorB',
                    version: '11',
                    total: 20
                },
                {
                    vendor: 'VendorB',
                    version: '8',
                    total: 15
                }
            ]
        };

        const wrapper = shallow(<JavaRuntimesCard reportWorkloadSummary={reportWorkloadSummary} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('expect to render empty card when undefined reportWorkloadSummary', () => {
        const wrapper = shallow(<JavaRuntimesCard reportWorkloadSummary={null} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('expect to render empty card when no java runtimes', () => {
        const reportWorkloadSummary: ReportWorkloadSummary = {
            ...BasicReportWorkloadSummary,
            javaRuntimes: []
        };

        const wrapper = shallow(<JavaRuntimesCard reportWorkloadSummary={reportWorkloadSummary} />);
        expect(wrapper).toMatchSnapshot();
    });
});
