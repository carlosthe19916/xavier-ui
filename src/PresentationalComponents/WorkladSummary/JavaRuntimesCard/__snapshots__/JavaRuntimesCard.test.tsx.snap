// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaRuntimesCard expect to render 1`] = `
<ReportCard
  skipBullseye={true}
  title={
    <span>
      <span>
        Oracle Java runtime information
      </span>
       
      <span>
        <Tooltip
          appendTo={[Function]}
          aria="describedby"
          boundary="window"
          className=""
          content={
            <div>
              See the Workload migration inventory for details
            </div>
          }
          distance={15}
          enableFlip={true}
          entryDelay={500}
          exitDelay={500}
          flipBehavior={
            Array [
              "top",
              "right",
              "bottom",
              "left",
              "top",
              "right",
              "bottom",
            ]
          }
          id=""
          isAppLauncher={false}
          isContentLeftAligned={false}
          isVisible={false}
          maxWidth="18.75rem"
          position="top"
          tippyProps={Object {}}
          trigger="mouseenter focus"
          zIndex={9999}
        >
          <HelpIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </Tooltip>
      </span>
    </span>
  }
>
  <Grid
    hasGutter={true}
    xl={6}
  >
    <GridItem>
      <Bullseye>
        <FancyChartDonut
          chartProps={
            Object {
              "subTitle": "Total",
              "title": "45",
            }
          }
          data={
            Array [
              Object {
                "extraData": 10,
                "label": "VendorA JDK 8",
                "value": 0.2222222222222222,
              },
              Object {
                "extraData": 15,
                "label": "VendorB JDK 8",
                "value": 0.3333333333333333,
              },
              Object {
                "extraData": 20,
                "label": "VendorB JDK 11",
                "value": 0.4444444444444444,
              },
            ]
          }
          tickFormat={[Function]}
          tooltipFormat={[Function]}
        />
      </Bullseye>
    </GridItem>
    <GridItem>
      <Bullseye>
        <SolidCard
          title="Oracle JDKs that can be replaced with Open JDK: 8"
          width={510}
        />
      </Bullseye>
    </GridItem>
  </Grid>
</ReportCard>
`;

exports[`JavaRuntimesCard expect to render empty card when no java runtimes 1`] = `
<Component
  cardTitle={
    <span>
      <span>
        Oracle Java runtime information
      </span>
       
      <span>
        <Tooltip
          appendTo={[Function]}
          aria="describedby"
          boundary="window"
          className=""
          content={
            <div>
              See the Workload migration inventory for details
            </div>
          }
          distance={15}
          enableFlip={true}
          entryDelay={500}
          exitDelay={500}
          flipBehavior={
            Array [
              "top",
              "right",
              "bottom",
              "left",
              "top",
              "right",
              "bottom",
            ]
          }
          id=""
          isAppLauncher={false}
          isContentLeftAligned={false}
          isVisible={false}
          maxWidth="18.75rem"
          position="top"
          tippyProps={Object {}}
          trigger="mouseenter focus"
          zIndex={9999}
        >
          <HelpIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </Tooltip>
      </span>
    </span>
  }
  description="No instances of Oracle JDKs have been discovered."
  message="No instances found"
/>
`;

exports[`JavaRuntimesCard expect to render empty card when undefined reportWorkloadSummary 1`] = `
<Component
  cardTitle={
    <span>
      <span>
        Oracle Java runtime information
      </span>
       
      <span>
        <Tooltip
          appendTo={[Function]}
          aria="describedby"
          boundary="window"
          className=""
          content={
            <div>
              See the Workload migration inventory for details
            </div>
          }
          distance={15}
          enableFlip={true}
          entryDelay={500}
          exitDelay={500}
          flipBehavior={
            Array [
              "top",
              "right",
              "bottom",
              "left",
              "top",
              "right",
              "bottom",
            ]
          }
          id=""
          isAppLauncher={false}
          isContentLeftAligned={false}
          isVisible={false}
          maxWidth="18.75rem"
          position="top"
          tippyProps={Object {}}
          trigger="mouseenter focus"
          zIndex={9999}
        >
          <HelpIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </Tooltip>
      </span>
    </span>
  }
  description="No instances of Oracle JDKs have been discovered."
  message="No instances found"
/>
`;
